hash_practice/exercises.py

```diff
@@ -2,18 +2,57 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(m * nlogn) where m is the length of strings and n is the length of the longest string in strings
+        Space Complexity: O(n)
     """
-    pass
+    sorted_strings = {}
+    output = []
+
+    for s in strings:
+        s_list = list(s)
+        s_list.sort()
+        sorted_s = "".join(s_list)
+        if sorted_s in sorted_strings:
+            index = sorted_strings[sorted_s]
+            output[index].append(s)
+        else:
+            sorted_strings[sorted_s] = len(output)
+            output.append([s])
+
+    return output
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n log(n))
+        Space Complexity: O(n)
     """
-    pass
+    if not nums:
+        return []
+    
+    freq_dict = {}
+
+    for n in nums:
+        if n in freq_dict:
+            freq_dict[n] += 1
+        else:
+            freq_dict[n] = 1
+
+    l = []
+    
+    for key, val in freq_dict.items():
+        l.append({'number': key, 'freq': val})
+    l.sort(key=sort_by_freq, reverse=True)
+
+    most_freq = []
+
+    for i in range(k):
+        most_freq.append(l[i]['number'])
+
+    return most_freq
+
+def sort_by_freq(dict):
+    return dict['freq']
 
 
 def valid_sudoku(table):

```

Comment:
✨ However, since the table will always be the same size, we can say time and space complexity are O(1)